internal/observability/call.go

```diff
@@ -59,14 +59,25 @@ func (c call) EndWithAppError(err error, isApplicationError bool) {
 }
 
 func (c call) endLogs(elapsed time.Duration, err error, isApplicationError bool) {
-	msg := "Handled inbound request."
-	if c.direction != _directionInbound {
-		msg = "Made outbound call."
+	var ce *zapcore.CheckedEntry
+	if err == nil {
```

Comment:
Sorry for the late review - there's a blocking problem here. We'll need to amend this check to `err == nil && !isApplicationError` to distinguish to correctly distinguish errors from successes. (You can see the same logic below on line 84.)

This is yet another place where having Thrift model errors as successful responses with a binary payload is painful. From the perspective of the network transport - which is where YARPC middleware operates - Thrift exceptions are successful responses with a special binary payload. The `err` parameter here will only capture lower-level network errors, not application errors.